gordo_components/data_provider/iroc_reader.py

```diff
@@ -59,14 +59,12 @@ def load_series(
             map(lambda tag: IrocReader.base_path_from_asset(tag.asset), tag_list)
         )
         if len(set(base_paths_from_assets)) > 1:
-            logger.warning(
-                "Iroc reader found more than one asset from the tag list provided, returning none."
+            raise ValueError(
+                "Iroc reader found more than one asset from the tag list provided"
             )
             return
-        elif len(base_paths_from_assets) != len(tag_list):
-            logger.warning(
-                "Iroc reader could not associate some tags to an asset, returning none."
-            )
+        elif any([path is None for path in base_paths_from_assets]):
```

Comment:
Here you can rather write `elif None in base_paths_from_assets:`

Reply:
Done!